### DISCUSSION TITLE
next-i18next with styled-component

### DISCUSSION THREAD

**Discussion by ffan0811** [i18next/next-i18next#1485] [Q&A] [answered]
I'm trying to upgrade Next.js from `v^9.3.4` to `v10.2.0` and next-i18next from `v^6.0.2` to `v8.9.0`.

I read README of next-i18next that it doesn't support `getInitialProps` anymore after v8.0.0

I use `getInitialProps` for styled-components. To use styled-components, It requires me to use `getInitialProps`. (reference: https://github.com/vercel/next.js/blob/master/examples/with-styled-components/pages/_document.js)

What should I do now?
I don't want to give up on styled-components or Next.js `v10.2.0`.

**Comment by isaachinman** [comment, marked as answer]
Well, I think we've got two obvious solutions:

1. Start supporting `getInitialProps` in `next-i18next`
2. Drop requirement for `getInitialProps` in `styled-components`

Bear in mind that:

> getInitialProps will disable Automatic Static Optimization.

I'd recommend you investigate the issue on the NextJs/`styled-components` side, as there are probably more users of `styled-components` than `next-i18next`, and requiring `getInitialProps` is requiring all users to take a big performance hit.

This may be relevant:

https://github.com/vercel/next.js/discussions/22065
